src/App.tsx

```diff
@@ -1,15 +1,63 @@
-import React from 'react';
+import React, { useMemo, useRef, useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
-export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+type Props = {
+  delay?: number;
+}
+
+export const App: React.FC<Props> = ({ delay = 300 }) => {
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [showSuggestions, setShowSuggestions] = useState(false);
+  const [selectPerson, setSelectPerson] = useState<Person | null>(null);
+  const [highlightedIndex, setHighlightedIndex] = useState<number | null>(null);
+
+  const timer = useRef(0);
+
+  const filteredPeople = useMemo(() => {
+    const trimmedQuery = appliedQuery.trim();
+
+  if (trimmedQuery === '') {
+    return peopleFromServer;
+  }
+  return peopleFromServer.filter(people =>
+    people.name.toLowerCase().includes(trimmedQuery.toLowerCase())
+  );
```

Comment:
The filter function should not run if the input contains only spaces. You should check if `trimmedQuery` is not empty before filtering the `peopleFromServer` array. This aligns with the checklist requirement: 'make sure that filter won't be called if user entered spaces only'.